### PR TITLE
Update Jetpack button and card styles to match WordPress 5.3

### PR DESCRIPTION
**Pull request by gravityrail**
This is just an experiment I did to make the Jetpack buttons and borders look a bit closer to the "flat" styles coming in WordPress 5.3. They may, as a result, look a bit out of place on older versions of WordPress, but they _already_ don't match those versions, so I think this is a net win.

I am not a designer, so I was really just eyeballing this and trying to make it look "closer", not perfect.

This also lays the groundwork for integrating `@wordpress/component`s into our Jetpack admin pages without a style mismatch.

#### Changes proposed in this Pull Request:

* Update Jetpack styles to more closely match WordPress 5.3 styles.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Just slightly modifies styles

#### Proposed changelog entry for your changes:

* Update UI styles to fit in WordPress 5.3 core components

#### Screenshots

Here's a cropped before/after shot.

Before:

myplan-current-cropped

After:

myplan-53-cropped

A gallery of the WP 5.3 dash/posts, plus Jetpack before/after screenshots, is available here:

https://cloudup.com/czXdFjw6ykm

### CODE REVIEW

**Review by keoshi — DISMISSED** [review]
This is looking good! Below are some of the things I've noticed:

### Set up Jetpack button

Not sure exactly what's the scope here, so don't know if this applies, but the green Jetpack button in the initial pre-connection banner has a border that the primary buttons in 5.3 don't (or are the same color as the bg to be exact). It also has the existing Jetpack `:active` styles, which make it look weird.

jp-connect-button

### Buttons

`:focus` styles don't match what's in 5.3. Same goes for default buttons.

image

image

```
box-shadow:
    0 0 0 1px #5b9dd9,
    0 0 2px 1px rgba(30, 140, 190, 0.8);
```

`:active` styles don't match what's in 5.3. Same goes for default buttons.

image

image

`background: #00669B`

### Double borders

At least the Site Stats block appears to have a double border. It seems to be caused by a parent and child elements where both have a border property.

image

### Border radii

The change in border radius seems to have some side effects, where adjacent cards have an undesired dip inside where they should be flush instead.

image

### Margins

Some change in margins makes the external link detached from its card.

image

**Comment by MichaelArestad** [comment]
I did some pretty major changes:
* Updated button focus styles (only on primary/secondary)
* Removed the border radius on card
* Updated border colors on cards via box shadow (to avoid recoding everything)
* Updated card border colors on non-card components to match
* fixed a stats display bug (no more gap)

Suggestions for future updates and migration to WordPress components:
* One PR to import the components - then merge it in
* One PR per component - perhaps even more focused

I apologize for whoever has to review this, but it's ready for another review.

**Comment by MichaelArestad** [comment]
Scratch that. Found a bunch more styles that need updating in signup flows etc.

**Comment by MichaelArestad** [comment]
Should be good now. Items to review:
- [ ] Cards
- [ ] Buttons
- [ ] ButtonGroups
- [ ] Jetpack masthead
- [ ] Jetpack dashboard borders and buttons
- [ ] Jetpack settings
- [ ] Disconnected states
- [ ] JITM borders
- [ ] Settings group borders and buttons

**Comment by gravityrail** [comment]
Button style on the connection iframe doesn't match Jetpack's new style:

approve-button-style

Since this is rendered on WPCOM, we could switch out button styles based on the jetpack_version flag passed in via the iframe URL.

**Comment by gravityrail** [comment]
How do we feel about the iridescent green setup banner in this new style?

Might be out of scope for this ticket since the color is an unrelated style change.

setup-banner

**Comment by gravityrail** [comment]
Consider this comment a 👍 on this change (I can't leave an actual approval since I still own the PR)

**Comment by gravityrail** [comment]
@keoshi can you take another look and verify that @MichaelArestad fixed the issues you saw?

**Comment by MichaelArestad** [comment]
> How do we feel about the iridescent green setup banner in this new style?

We're going to leave it. At least for this PR.

**Comment by joanrho** [comment]
@MichaelArestad  Tested all of the following on WordPress 5.3-RC1-46561 running Twenty Twenty/twentytwenty-master theme:
 
- [x] Cards
- [x] Buttons
- [x] ButtonGroups
- [x] Jetpack masthead
- [x]  Jetpack dashboard borders and buttons
- [x]  Jetpack settings
- [x]  Disconnected states
- [x]  JITM borders
- [x]  Settings group borders and buttons

These are all looking great, but I caught a few minor things:
1. "Show me" button text on the Stats page is a bit off (See #13783)
2. `dops-section-header` height alignment could be evened out (See #13781)
3. (might be a core design issue) The border-radius of the "Add New" button on the tops of the Posts, Media, Pages, Themes, Plugins, pages is 2px whereas the styles everywhere else in 5.3 all are border-radius: 3px. Screenshot: 
Screen Shot 2019-10-18 at 2 42 35 PM

**Comment by keoshi** [comment]
Can confirm that number 3 is a core issue that’s currently being worked on!

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good to me, but we'll need to add some of those CSS files back.

I also have a few other questions and remarks, but I don't think any of them are blockers.

*******

Should the non-active color be bright blue like this? 

image
image

It does not seem in line with the rest of the CSS in wp-admin:

image

*******************

It seems there is a bit of padding missing here:
image

******

Maybe that's just me, but it feels odd to go from the closed dropdown with shadows to the flat borders of the open dropdown:

Screen Recording 2019-10-23 at 02 21 PM

*********

The blue border around the input fields seem to be bright blue. Is that the same color as Core?

image

File: packages/jitm/assets/jetpack-admin-jitm-rtl.css [hunk 1]
```diff
@@ -1,479 +0,0 @@
-/*!
```
[the remark this answers is not in the document]
Comment (MichaelArestad) [reply packages/jitm/assets/jetpack-admin-jitm-rtl.css]: Hmm. I thought these were generated during build so we didn't want to sync them. I didn't specifically remove this one (at least not intentionally). I'll add it back.

[the remark this answers is not in the document]
Comment (MichaelArestad) [reply packages/jitm/assets/jetpack-admin-jitm-rtl.css]: Added

**Review by jeherve — APPROVED** [review]
Looking good now. Merging.

**Pull request merged**